File: sorting_algorithms/bubble_sort.rs
fn bubble_sort(a: &mut [i32]) {
    
    // to len()-1 cause we gonna use this to get the position in array
    for i in 0..a.len()-1 {
        // to len()-2 cause we will use this to get the adjacent element, +1 
        for k in 0..a.len()-1 {
            println!("index {}",k);
            if a[k] > a[k+1] {
                let temp = a[k];
                a[k] =  a[k+1];
                a[k+1] = temp;
            }
        }
    }
}

fn main() {
    let mut array = [2,7,4,1,3];
    bubble_sort(&mut array);
    println!("{:?} {}",array,array.len());
    
    // here loop does not go till 4, only till three
    for i in 0..4 {
        println!("yo {}",i);
    }
}

File: sorting_algorithms/bubble_sort/bubble_sort.go
package main

import (
	"fmt"
)

func bubbleSort(array []int) []int {
	for i := range array {
		fmt.Println(i)
		for k := 0; k < len(array)-1; k++ {
			if array[k] > array[k+1] {
				temp := array[k]
				array[k] = array[k+1]
				array[k+1] = temp
			}
		}
	}
	return array
}

func main() {
	s := []int{2, 7, 4, 1, 3}
	fmt.Println(bubbleSort(s))
}
